util_test.ts

```diff
@@ -243,10 +235,132 @@ Deno.test("cacheDir", async ({ step }) => {
+
+      Deno.env.delete("XDG_CACHE_HOME");
+    });
+
+    await step("HOME", () => {
+      Deno.env.set("HOME", cache);
+
+      const dir = cacheDir();
+      assert(dir);
+      assertEquals(basename(dir), ".cache");
+      assertEquals(basename(dirname(dir)), "cache");
+      assertEquals(basename(dirname(dirname(dir))), "plug");
+
+      Deno.env.delete("HOME");
+    });
+  });
+
+  await step("solaris", async ({ step }) => {
+    // @ts-ignore TS2540
+    Deno.build = { os: "solaris" };
+
+    await step("XDG_CACHE_HOME", () => {
+      Deno.env.set("XDG_CACHE_HOME", cache);
+
+      const dir = cacheDir();
+      assertEquals(dir, cache);
+
+      Deno.env.delete("XDG_CACHE_HOME");
+    });
+
+    await step("HOME", () => {
+      Deno.env.set("HOME", cache);
+
+      const dir = cacheDir();
+      assert(dir);
+      assertEquals(basename(dir), ".cache");
+      assertEquals(basename(dirname(dir)), "cache");
+      assertEquals(basename(dirname(dirname(dir))), "plug");
+
+      Deno.env.delete("HOME");
+    });
+  });
+
+  await step("netbsd", async ({ step }) => {
+    // @ts-ignore TS2540
+    Deno.build = { os: "netbsd" };
+    await step("XDG_CACHE_HOME", () => {
+      Deno.env.set("XDG_CACHE_HOME", cache);
+
+      const dir = cacheDir();
+      assertEquals(dir, cache);
+
+      Deno.env.delete("XDG_CACHE_HOME");
+    });
+
+    await step("HOME", () => {
+      Deno.env.set("HOME", cache);
+
+      const dir = cacheDir();
+      assert(dir);
+      assertEquals(basename(dir), ".cache");
+      assertEquals(basename(dirname(dir)), "cache");
+      assertEquals(basename(dirname(dirname(dir))), "plug");
+
+      Deno.env.delete("HOME");
+    });
+  });
+
+  await step("aix", async ({ step }) => {
+    // @ts-ignore TS2540
+    Deno.build = { os: "aix" };
+
+    await step("XDG_CACHE_HOME", () => {
+      Deno.env.set("XDG_CACHE_HOME", cache);
+
+      const dir = cacheDir();
+      assertEquals(dir, cache);
+
+      Deno.env.delete("XDG_CACHE_HOME");
+    });
+
+    await step("HOME", () => {
+      Deno.env.set("HOME", cache);
+
+      const dir = cacheDir();
+      assert(dir);
+      assertEquals(basename(dir), ".cache");
+      assertEquals(basename(dirname(dir)), "cache");
+      assertEquals(basename(dirname(dirname(dir))), "plug");
+
+      Deno.env.delete("HOME");
+    });
+  });
+
+  await step("illumos", async ({ step }) => {
+    // @ts-ignore TS2540
+    Deno.build = { os: "illumos" };
+    await step("XDG_CACHE_HOME", () => {
+      Deno.env.set("XDG_CACHE_HOME", cache);
+
+      const dir = cacheDir();
+      assertEquals(dir, cache);
+
+      Deno.env.delete("XDG_CACHE_HOME");
+    });
+
+    await step("HOME", () => {
+      Deno.env.set("HOME", cache);
+
+      const dir = cacheDir();
+      assert(dir);
+      assertEquals(basename(dir), ".cache");
+      assertEquals(basename(dirname(dir)), "cache");
+      assertEquals(basename(dirname(dirname(dir))), "plug");
+
+      Deno.env.delete("HOME");
+    });
```

Comment:
We don't need separate tests for all separate unix OS's if their behaviour does not differ. But they might actually differ so that might be worth looking into.

Reply:
lets keep that for now since if we find out they do we can change that